### PR TITLE
[Bug](pipeline) Fix pipeline load lose data use wrong var check order

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HappenLee** [comment]
run buildall

File: be/src/vec/sink/writer/async_result_writer.cpp [hunk 1]
```diff
@@ -68,9 +66,10 @@ Status AsyncResultWriter::sink(Block* block, bool eos) {
             _dependency->block();
         }
     }
+    _eos = eos;
```
Comment (yiguolei) [inline be/src/vec/sink/writer/async_result_writer.cpp:69]: need some comment here, to explain why this will fix the bug

**Comment by HappenLee** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**